tool/firebase_util.py

```diff
@@ -0,0 +1,12 @@
+
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import firestore
+
+def init_firebase_client(CRYPTO_TOKEN_PATH, log):
```

Comment:
In general, none of the code passes a `log` parameter, so it feels out of place here.
Consider something like:
```
log = None

def init_firebase_client(CRYPTO_TOKEN_PATH):
    global log
    if log is None:
        log = logging.Logger(__file__, CRYPTO_TOKEN_PATH)
    ... etc ...
```

Reply:
They actually all pass a log parameter. Initially had done something similar to this but after some suggestions opted to pass in the log parameter since not all callers are using the same logger

Reply:
I've understood the reasoning behind this. I've removed the passing of logging into this function